pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -107,7 +107,16 @@ func (configgen *ConfigGeneratorImpl) BuildClusters(env *model.Environment, prox
 
 	// Add a blackhole and passthrough cluster for catching traffic to unresolved routes
 	// DO NOT CALL PLUGINS for these two clusters.
-	clusters = append(clusters, buildBlackHoleCluster(env), buildDefaultPassthroughCluster(env))
+	inboundPassthroughCluster := buildDefaultPassthroughCluster(env)
+	inboundPassthroughCluster.UpstreamBindConfig = &core.BindConfig{
+		SourceAddress: core.SocketAddress{
```

Comment:
Please create a separate function. This cluster is also a duplicate.

Reply:
Sorry for the noise! This experimental commits are not supposed to be pushed here for review :(

Thanks for the early comment!